### DISCUSSION TITLE
invalid input text "add"

### DISCUSSION THREAD

**Discussion by andy-maier** [mikefarah/yq#1957] [Q&A]
I am trying to use the "add" filter to add the items in an array of integers, but that results in 'invalid input text "add"':

```
$ echo "
- 1
- 2
" | yq add
Error: 1:1: invalid input text "add"

$ echo "[1, 2]" | yq add
Error: 1:1: invalid input text "add"
```

According to the jq documentation for "add", this ought to work: https://jqlang.github.io/jq/manual/#add

```
$ echo "[1, 2]" | jq add
3
```

yq version: 4.35.2
jq version: 1.7
os: macos
shell: bash

Is this a bug?
How do I need to phrase the yq command so that the addition works?

**Update:**

After reading some more, it seems that the "yq" command simply does not support an "add" filter. Instead it supports the "+" operator. But the documentation has no example on how "+" can be used to add the integers in an array.

What I use right now, is:
```
$ echo "
- 1
- 2
" | yq -o json | jq add
3
```

This is pretty disappointing from a yq perspective, I would say. So if there is a yq-only solution, please post it here (and update the yq documentation with such an example).

**Comment by mikefarah** [comment]
I haven't yet added `add` (though I'd take a PR for it). `add` however is just syntactic sugar for summing with reduce, see: https://mikefarah.gitbook.io/yq/operators/reduce#sum-numbers
